homeassistant/components/renault/pyzeproxy.py

```diff
@@ -0,0 +1,112 @@
+"""Proxy to handle account communication with Renault servers via PyZE."""
+import asyncio
+import logging
+
+from pyze.api import BasicCredentialStore, Gigya, Kamereon, Vehicle
+
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+
+from .const import (
+    CONF_GIGYA_APIKEY,
+    CONF_KAMEREON_ACCOUNT_ID,
+    CONF_KAMEREON_APIKEY,
+    CONF_LOCALE,
+)
+from .pyzevehicleproxy import PyzeVehicleProxy
+
+LOGGER = logging.getLogger(__name__)
+
+
+class PyzeProxy:
+    """Handle account communication with Renault servers via PyZE."""
+
+    def __init__(self, hass):
+        """Initialise proxy."""
+        LOGGER.debug("Creating PyzeProxy")
+        self._hass = hass
+        self._gigya = None
+        self._kamereon = None
+        self._vehicle_links = None
+        self._vehicle_proxies = {}
+        self._vehicles_lock = asyncio.Lock()
```

Comment:
never reassigned ?

Reply:
I don't understand what you mean about this comment.
The lock object is created during `__init__`, and then used in `get_vehicle_proxy` to avoid loading multiple loads.